security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/ConditionSerializer.java

```diff
@@ -41,12 +41,33 @@ public void serialize(Condition condition, JsonGenerator jsonGenerator, Serializ
             case AnyViolationCondition.NAME:
                 serializeFilters((AbstractFilteredCondition) condition, jsonGenerator, serializerProvider);
                 break;
+            case InjectionThresholdCondition.NAME:
+                serializeThresholdCondition((AbstractThresholdCondition) condition, jsonGenerator);
+                break;
+            case BranchThresholdCondition.NAME:
+                BranchThresholdCondition branchCondition = (BranchThresholdCondition) condition;
+                serializeThresholdCondition(branchCondition, jsonGenerator);
+                jsonGenerator.writeStringField("side", branchCondition.getSide().name());
+                break;
+            case AcDcConverterThresholdCondition.NAME:
+                AcDcConverterThresholdCondition acDcConverterCondition = (AcDcConverterThresholdCondition) condition;
+                serializeThresholdCondition(acDcConverterCondition, jsonGenerator);
+                jsonGenerator.writeBooleanField("acSide", acDcConverterCondition.isAcSide());
+                jsonGenerator.writeNumberField("terminalNumber", acDcConverterCondition.getTerminalNumber());
+                break;
             default:
                 throw new IllegalArgumentException("condition type \'" + condition.getType() + "\' does not exist");
         }
         jsonGenerator.writeEndObject();
     }
 
+    public void serializeThresholdCondition(AbstractThresholdCondition condition, JsonGenerator jsonGenerator) throws IOException {
+        jsonGenerator.writeNumberField("threshold", condition.getThreshold());
+        jsonGenerator.writeStringField("equipmentId", condition.getEquipmentId());
+        jsonGenerator.writeStringField("comparisonType", condition.getComparisonType().name());
+        jsonGenerator.writeStringField("variable", condition.getVariable().name());
```

Comment:
Maybe the order can be changed to have a more natural reading order (even thought the JSON file is not primarily made for reading it):
- equipmentId
- variable
- comparisonType
- threshold